packages/@aws-cdk/core/lib/app.ts

```diff
@@ -134,6 +147,16 @@ export class App extends Construct {
     return assembly;
   }
 
+  protected prepare(): void {
+    if (this._prepared) {
+      return;
+    }
+    if (this.constructTreeMetadata) {
+      new ConstructTreeMetadata(this);
```

Comment:
Why not add this in the constructor? What's the value in doing this in `prepare`?

Reply:
Had an offline conversation on this. I'll be moving this into the constructor.